packages/components/src/client/utils/mse.ts

```diff
@@ -0,0 +1,119 @@
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+
+export const autoType = (url: string): string | undefined => {
+  if (url) {
+    switch (true) {
+      case url.endsWith(".m3u8"):
+        return "m3u8";
+      case url.endsWith(".flv"):
+        return "flv";
+      case url.endsWith(".mpd"):
+        return "mpd";
+      default:
+        return "mp4";
+    }
+  }
+
+  return undefined;
+};
```

Comment:
> I do not think this function is ok to me, you should get the actual ext and switch that.

When using a simple configuration such as

```html
<ArtPlayer url="/assets/video/dash/master.m3u8" />
```

not

```html
<ArtPlayer url="/assets/video/dash/master.m3u8" type="hls" />
```
This can greatly simplify the steps to use

Reply:
I am talking about code quality. Extrat the ext from url,  and switch that value.

Reply:
> 



> I am talking about code quality. Extrat the ext from url, and switch that value.

OK, I know. I have a new idea, match the extension name to get the type, then compare the type list, and throw a warning if it is not in the list.